### PR TITLE
style: removed logger.error & improved error message

### PR DESCRIPTION
**Pull request by Alex-Werner**
### Issue being fixed or implemented  

This PR removes some logged errors and improved the Error message that is being thrown at the users.

### What was done  

- style: removed many logger.error
- style: improved the message that is thrown to the user on failure. 

### How Has This Been Tested?

Relying on current test suite. 

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by Alex-Werner** [comment]
Rebased onto master

File: examples/workers/ColdStorageWorker.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ class ColdStorageWorker extends Worker {
     const { walletConsolidator } = this;
     const utxos = this.getUTXOS();
     if (utxos.length === 0) {
-      logger.error('ColdStorageWorker : We did not found any utxos. Doing nothing');
+      throw new Error('ColdStorageWorker : We did not found any utxos. Doing nothing');
```
Comment (shumkov) [inline examples/workers/ColdStorageWorker.js:25]: Don't we want to introduce specific error classes?

Comment (Alex-Werner) [reply examples/workers/ColdStorageWorker.js:25]: It's actually not an error to not find a UTXO afaik, that's why it just tell you about it and continue process.

**Comment by Alex-Werner** [comment]
Thank you for review. 
@dalibran We decided that we should improve those generic error into some domain based one. Can I let you add that to our backlog ?

**Pull request merged**